File: android/app/src/main/kotlin/com/goalmaker/app/application/planning/HabitPause.kt
package com.goalmaker.app.application.planning

import java.time.LocalDate

/** Days a habit rests, [from] to [until] (null while the pause lasts): they neither break its streak nor count. */
data class HabitPause(
    val id: String,
    val habitId: String,
    val from: LocalDate,
    val until: LocalDate? = null,
    val deleted: Boolean = false,
)

File: windows/src/GoalMaker.Core/Planning/HabitPause.cs
namespace GoalMaker.Core.Planning;

/// <summary>Days a habit rests, <see cref="From"/> to <see cref="Until"/> (null while the pause lasts): they neither break its streak nor count.</summary>
public sealed record HabitPause(string Id, string HabitId, DateOnly From, DateOnly? Until = null, bool Deleted = false);
